### PR TITLE
[SDK-2708] Use configured domain for client assets download

### PR DESCRIPTION
**Pull request by stevehobbsdev**
### Changes

This PR reconfigures Lock to use the configured domain when constructing the URL to download the client assets file, instead of using `cdn.[locality].auth0.com`.

The change applies to:

* Configuring Lock with an Auth0 domain: `example.us.auth0.com`
* Custom domains: `auth.custom-domain.com`
* Auth0 domain with `__useTenantInfo: true` in the config
* Custom domain with `__useTenantInfo: true` in the config

Test coverage for these settings has been added.

### Testing

* [X] This change adds unit test coverage
* [ ] This change adds integration test coverage
* [X] This change has been tested on the latest version of the platform/language

### Checklist

* [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
* [X] All code quality tools/guidelines have been run/followed
* [X] All relevant assets have been compiled

### CODE REVIEW

**Review by joseluisdiaz — DISMISSED** [review]
LGTM!

File: src/__tests__/core/index.test.js [hunk 1]
```diff
@@ -47,19 +49,199 @@ describe('setup', () => {
...
+    it('should use the clientBaseUrl option if given', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {
+          clientBaseUrl: 'https://client-base-url.example.com',
+          configurationBaseUrl: 'https://config-base-url.example.com',
```
Comment (lbalmaceda) [inline src/__tests__/core/index.test.js:97]: this key and the above one have the same value. How can I, just by reading this one test, know that the value you end up getting in the end comes from "clientBaseUrl" and not "configurationBaseUrl"?

File: src/__tests__/core/index.test.js [hunk 2]
```diff
@@ -47,19 +49,199 @@ describe('setup', () => {
...
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://assets-url.example.com');
+    });
+  });
+
+  describe('tenantBaseUrl', () => {
+    it('tenantBaseUrl should default to domain URL when using auth0.com', () => {
```
Comment (lbalmaceda) [inline src/__tests__/core/index.test.js:156]: what does "domain URL" mean here? also, you might want to skip repeating yourself "tenantBaseUrl" at the beginning of the text name, as this is inside a describe already scoping that.

File: src/__tests__/core/index.test.js [hunk 3]
```diff
@@ -47,19 +49,199 @@ describe('setup', () => {
...
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.tenantBaseUrl).toBe('https://auth.my-tenant.com/info-v1.js');
```
Comment (lbalmaceda) [inline src/__tests__/core/index.test.js:195]: why the path differs in this one? what's the condition? you might want to reflect that in the test name

File: src/__tests__/core/index.test.js [hunk 4]
```diff
@@ -47,19 +49,199 @@ describe('setup', () => {
...
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.tenantBaseUrl).toBe('https://my-tenant.us.auth0.com/tenants/v1/my-tenant.js');
```
Comment (lbalmaceda) [inline src/__tests__/core/index.test.js:174]: "tenant base URL" is then compared against a URL that also has a complete path to a file (tenant.js), so might not really represent a "base URL"

File: src/__tests__/core/index.test.js [hunk 5]
```diff
@@ -47,19 +49,199 @@ describe('setup', () => {
...
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://assets-url.example.com');
+    });
+  });
+
+  describe('tenantBaseUrl', () => {
```
Comment (lbalmaceda) [inline src/__tests__/core/index.test.js:155]: all the tests here pass `__useTenantInfo: true`. What happens when that's false? (might be a missing test case)

**Review by lbalmaceda — CHANGES_REQUESTED** [review]

File: src/__tests__/core/index.test.js (hunk 1 above)
Comment (stevehobbsdev) [reply src/__tests__/core/index.test.js:97]: The values are different. Similar, I admit, but different. I can change it so it's more obvious 👍🏻

File: src/__tests__/core/index.test.js (hunk 2 above)
Comment (stevehobbsdev) [reply src/__tests__/core/index.test.js:156]: Thanks, I had meant to change that 👍🏻

File: src/__tests__/core/index.test.js (hunk 4 above)
Comment (stevehobbsdev) [reply src/__tests__/core/index.test.js:174]: I agree, the naming isn't the best. However, the name exists from way before these changes,

File: src/__tests__/core/index.test.js (hunk 5 above)
Comment (stevehobbsdev) [reply src/__tests__/core/index.test.js:155]: If `__useTenantInfo` is false or not supplied, the code that figures out the client settings URL based on the `configurationBaseUrl` value, which is what the other tests are concerned with.

File: src/__tests__/core/index.test.js (hunk 1 above)
Comment (lbalmaceda) [reply src/__tests__/core/index.test.js:97]: Rereading this, I see now they are different. It's OK to keep them.

File: src/__tests__/core/index.test.js (hunk 3 above)
Comment (stevehobbsdev) [reply src/__tests__/core/index.test.js:195]: Yeah I'm not sure of the history as to why this is. If you use a custom domain, the URL differs when using `__useTenantInfo` vs using an `auth0.com` domain. Just trying to keep and verify the same logic here.

**Review by lbalmaceda — APPROVED** [review]

**Pull request merged**